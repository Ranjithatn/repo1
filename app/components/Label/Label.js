import React from 'react'

const Label = ({
  htmlFor,
  text,
  name,
  style={},
  disabled=false,
}) => {
  let disabledStyle = {};
  if ( disabled ) { disabledStyle = { color: "rgba(0,0,0,0.45)" }; }
   return (
    <label className="label" htmlFor={htmlFor ? htmlFor : ""} name={name ? name : ""} style={{...style, ...disabledStyle}}>
      { text }
    </label>
   )
}

export default Label;
