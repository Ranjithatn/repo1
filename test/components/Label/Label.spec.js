import React from "react";
import toJson from "enzyme-to-json";
import Label from "../../../app/components/Label/Label";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });


describe('<Label />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <Label  />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
