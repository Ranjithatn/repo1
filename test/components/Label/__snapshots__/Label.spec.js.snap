// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render without crashing 1`] = `
<Label>
  <label
    className="label"
    htmlFor=""
    name=""
    style={Object {}}
  />
</Label>
`;
